src/transformers/models/gpt2/modeling_gpt2.py

```diff
@@ -218,63 +231,72 @@ def forward(
         if encoder_hidden_states is not None:
             assert hasattr(
                 self, "q_attn"
-            ), "If class is used as cross attention, the weights `q_attn` have to be defined. Please make sure to instantiate class with `Attention(..., is_cross_attention=True)`."
+            ), "If class is used as cross attention, the weights `q_attn` have to be defined. Please make sure to instantiate class with `GPT2Attention(..., is_cross_attention=True)`."
```

Comment:
This is not super readable with the long error message, so would go for the syntax
```
if not hasattr(self, "q_attn"):
    raise ValueError(...)
```
and the error message can be split into several lines.